pkg/services/playlist/playlistimpl/xorm_store.go

```diff
@@ -13,22 +14,26 @@ type sqlStore struct {
 	db db.DB
 }
 
+var _ store = &sqlStore{}
+
 func (s *sqlStore) Insert(ctx context.Context, cmd *playlist.CreatePlaylistCommand) (*playlist.Playlist, error) {
 	p := playlist.Playlist{}
-	err := s.db.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
-		uid, err := generateAndValidateNewPlaylistUid(sess, cmd.OrgId)
-		if err != nil {
-			return err
-		}
+	if cmd.UID == "" {
+		cmd.UID = util.GenerateShortUID()
+	}
```

Comment:
I'm not sure where the k8s-specifical validation should occur, but we need to validate any user-supplied UID before storing in legacy storage:

https://github.com/grafana/grafana/blob/2a6c16d9699152688029f73ad80b8fe974f55481/pkg/util/shortid_generator.go#L65-L66

Reply:
Added tests here https://github.com/grafana/grafana/pull/76295/files#diff-450ec0bb14680d84de7a49e86795200b078d6112a9814d1c1cd6f864d1c56d1fR94